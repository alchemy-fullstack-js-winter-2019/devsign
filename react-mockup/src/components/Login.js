import React from 'react';

function Login() {
  return (
    <section>
      <h2>Log-in</h2>
      <label htmlFor="username">Username:</label>
      <input type="text" name="username" />
      <label htmlFor="password">Password:</label>
      <input type="password" name="password" />
      <button>Sign-in</button>
    </section>
  );
}

export default Login;
