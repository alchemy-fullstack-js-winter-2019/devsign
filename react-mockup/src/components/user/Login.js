import React from 'react';
import { LoginSection } from './UserStyles';

function Login() {
  return (
    <LoginSection>
      <h2>Log-in</h2>
      <label htmlFor="username">Username:</label>
      <input type="text" name="username" />
      <label htmlFor="password">Password:</label>
      <input type="password" name="password" />
      <button>Sign-in</button>
    </LoginSection>
  );
}

export default Login;
