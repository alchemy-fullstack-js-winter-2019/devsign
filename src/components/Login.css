input {
  padding: .5em .6em;
  display: inline-block;
  border: 1px solid #ccc;
  box-shadow: inset 0 1px 3px #ddd;
  border-radius: 4px;
  vertical-align: middle;
  box-sizing: border-box;
  margin: 1em 0em;
}
.labels {
  margin: 0;
  text-align: center;
  display: inline;
	position: relative;
	font: 1em Helvetica, Sans-Serif;
	letter-spacing: 5px;
	color: rgba(255,0,0,0.5);
}
::placeholder {
  margin: 0;
  text-align: center;
  display: inline;
	position: relative;
	font: 1em Helvetica, Sans-Serif;
	letter-spacing: 2px;
	color: rgba(0,0,0,0.5);
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: -2em;
}

h1 {
  margin: 0;
  text-align: center;
  display: inline;
	position: relative;
	font: 5em Helvetica, Sans-Serif;
	letter-spacing: -5px;
	color: rgba(0,0,255,0.5);
}
h1:after {
	content: "Rockin Robin";
	position: absolute; left: 10px; top: 5px;
	color: rgba(255,0,0,0.5);   
}

button {
  margin: 0;
  text-align: center;
  display: inline;
	position: relative;
	font: 1em Helvetica, Sans-Serif;
	letter-spacing: 1px;
  color: rgba(0,0,255,0.5);
  margin-left: -.5em;
}



