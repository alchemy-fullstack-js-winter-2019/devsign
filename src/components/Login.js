import React from 'react';
import { Redirect } from 'react-router-dom';
import { ROUTES } from '../routes';

function Login() {
  
  return (
    
    <Redirect to={ROUTES.HOME.linkTo()} />
    // <section>
    //   <form>
    //     <h3>SIGN IN</h3>
    //     <label htmlFor="email">Email</label>
    //     <input type="email" name="email" id="email" size="30" placeholder="youremail@example.com" required/>
    //     <label htmlFor="password">Password</label>
    //     <input type="text" name="password" id="password" size="30" required/>
    //     <button>Submit</button>
    //   </form>
    // </section>
  );
}

export default Login;
