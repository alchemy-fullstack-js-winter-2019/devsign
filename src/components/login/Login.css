.Form {
    background-color: white;
    text-align: center;
    display: block;
    border: 2px black;
    border-radius: 1em;
    margin: .5em;
    padding: 1em;
    box-shadow: 2px 2px 2px 1px  black;
        label {
        text-align: center;
        font-size: 1.5em;
        display: block;
        input {
            align-content: center;
            display: block;
            border:  2px rgb(66, 66, 66) solid;
            border-radius: 2em;
        }
    }
    button {
        margin: 1em;
        font-size: 1.5em;
        background-color: white;
        border: 2px black;
        box-shadow: 2px 2px 2px 1px black;
    }
}