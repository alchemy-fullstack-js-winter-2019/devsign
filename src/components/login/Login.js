import React from 'react';
import PropTypes from 'prop-types';

function Login() {
  return (
    <section>
      <h2>Already a user?</h2>
      <figure>
        <img src="image" alt='user avatar' />
      </figure>
      <form>
        <label>Username or email</label>
        <input type="text" value='handle' />
        <label>Password</label>
        <input type="text" value='password' />
        <button>Tweet</button>
      </form>
    </section>
  );
}

Login.propTypes = {
  user: PropTypes.object
};
export default Login;
