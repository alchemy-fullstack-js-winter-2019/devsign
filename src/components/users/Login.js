import React, { Fragment } from 'react';
import Footer from '../app/Footer';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import ModalContainerSignup from '../../containers/ModalContainerSignup';

const Header = styled.header`
  display: flex;
  margin: 10px 0 10px 10px;
  padding-bottom: 10px;
  h1 {
      font-size: 3.5em;
      font-weight: 100;
      margin: 10px 10px 0 10px;
  }
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 40px;
  input {
    border: 1px solid darkgray;
    padding: 8px;
    border-radius: 7px;
    width: 200px;
    margin-bottom: 25px;
  }
  label {
    margin-bottom: 10px;
  }
  button {
    background: #00bfff;
    color: white;
    padding: 8px;
    border-radius: 7px;
    border: 1 px solid darkgray;
  }
`;

const Main = styled.main`
  background: #F5F5F5;
  margin: 0px;
  height: 100vh;
`;

const Wrapper = styled.section`
  text-align: center;
  margin-top: 50px;
  p {
    font-size: 1.5em;
  }
  
`;

const Aside = styled.aside`
  @media only screen and (max-width:600px)  { 
    display: none;
  }
`;
export default function Login() {
  return (
    <Fragment>
      <Header>
        <h1>Warble</h1>
        <img src="../assets/logo.png" height="80px" alt="logo"/>
      </Header>

      <Aside>
        <h2>JOIN WARBLE</h2>
        <ul>
          <li>Explore</li>
          <li>Connect</li>
          <li>Share</li>
        </ul>
      </Aside>

      <Main>
        <Form>
          <label>Username
            <input type="text" name="username" />
          </label>
          
          <label>Password
            <input type="text" name="password" />
          </label>
          
          <Link to='/home' name='home'><button type="submit">Login</button></Link>
        </Form>
        
        <Wrapper>
          <p>New user?</p>
          <ModalContainerSignup />
        </Wrapper>

      </Main>

      <Footer />
    </Fragment>
  );
}
