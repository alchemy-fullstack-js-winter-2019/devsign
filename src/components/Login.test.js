import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';
import { MemoryRouter } from 'react-router-dom';

describe('Login', () => {
  it('renders Login component', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot;
  });
});
