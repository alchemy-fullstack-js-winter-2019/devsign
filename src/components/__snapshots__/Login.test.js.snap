// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches a snapshot 1`] = `
<body>
  <header>
    <h1>
      Warble
    </h1>
    <img
      src="../assets/logo.png"
      width="40px"
    />
  </header>
  <aside>
    <h2>
      JOIN TWITTER
    </h2>
    <ul>
      <li>
        Explore
      </li>
      <li>
        Connect
      </li>
      <li>
        Share
      </li>
    </ul>
  </aside>
  <main>
    <label>
       Username
      <input
        type="text"
      />
    </label>
    <label>
       Password
      <input
        type="text"
      />
    </label>
    <button
      type="submit"
    >
      Login
    </button>
  </main>
  <footer>
    <p>
      Â© Warble | 2019 
    </p>
  </footer>
</body>
`;
